src/main/java/jenkins/plugins/openstack/compute/SlaveOptionsDescriptor.java

```diff
@@ -205,26 +210,28 @@ public ListBoxModel doFillHardwareIdItems(
             @QueryParameter String hardwareId,
             @QueryParameter String endPointUrl, @QueryParameter String identity, @QueryParameter String credential, @QueryParameter String zone
     ) {
-
         ListBoxModel m = new ListBoxModel();
         m.add("None specified", "");
-
+        final String valueOrEmpty = Util.fixNull(hardwareId);
+        boolean existingValueFound = valueOrEmpty.isEmpty();
         try {
-            final Openstack openstack = Openstack.Factory.get(endPointUrl, identity, credential, zone);
-            for (Flavor flavor : openstack.getSortedFlavors()) {
-                m.add(String.format("%s (%s)", flavor.getName(), flavor.getId()), flavor.getId());
+            if (haveServerDetails(endPointUrl, identity, credential, zone)) {
+                final Openstack openstack = Openstack.Factory.get(endPointUrl, identity, credential, zone);
+                for (Flavor flavor : openstack.getSortedFlavors()) {
+                    final String value = flavor.getId();
+                    final String displayText = String.format("%s (%s)", flavor.getName(), value);
+                    m.add(displayText, value);
+                    existingValueFound |= valueOrEmpty.equals(value);
+                }
             }
-            return m;
         } catch (AuthenticationException | FormValidation | ConnectionException ex) {
             LOGGER.log(Level.FINEST, "Openstack call failed", ex);
         } catch (Exception ex) {
             LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
         }
-
-        if (Util.fixEmpty(hardwareId) != null) {
+        if (!existingValueFound) {
```

Comment:
It's not that simple: `ListBoxModel` does not have a "contains a `ListBoxModel.Option` containing _value_" method.

`m.contains(...)` is simply the contains method from `ListBoxModel`'s parent class, `ArrayList<ListBoxModel.Option>`, so this method will only return `true` if we gave it a `ListBoxModel.Option`, and `ListBoxModel.Option` has no `equals(Object)` implementation so it will only ever return `true` if we gave it the exact same instance - a different instance containing the same contents would still result in `m.contains` returning false.
i.e. it's not as simple as just `m.contains(valueOrEmpty)` as that doesn't give you the answer you expect.

To remove the `existingValueFound` variable, we'd have to implement a new method like:
```
    private static boolean hasValue(ListBoxModel m, String value) {
        for( final ListBoxModel.Option o : m) {
            if ( Objects.equals(value, o.value)) return true;
        }
        return false;
    }
```
and then use `hasValue(m, valueOrEmpty)`.

So, given that, would you prefer `if (!existingValueFound) {` or `if (!hasValue(m, valueOrEmpty)) {` ?